### PR TITLE
Remove unneeded include files

### PR DESCRIPTION
**Pull request by TerranCivilian**
CLion's code inspection raised a few false alarms that broke compilation; I had to painstakingly weed them out. Everything else was removed without issue.

### CODE REVIEW

File: libyul/optimiser/LoadResolver.cpp [hunk 1]
```diff
@@ -30,8 +30,6 @@
 #include <libyul/AST.h>
 #include <libyul/Utilities.h>
 
-#include <libevmasm/Instruction.h>
-#include <libevmasm/GasMeter.h>
```
Comment (cameel) [inline libyul/optimiser/LoadResolver.cpp:34]: This file does use `GasMeter` though. It must be included through some other header but I think it should stay here too.

**Review by cameel — COMMENTED** [review]
Looks good!

At first I thought this might be removing a bit too much, i.e. headers for stuff that's actually used in the file but also included via another header. In practice I only found one instance of this (`GasMeter.h`) and most of the removed includes seem pretty generic so I think it's a net improvement even if there happen to be more cases like this.

**Review by Marenz — COMMENTED** [review]
LGTM. Will change to approved when cameels comments are taken care of :)

File: libyul/optimiser/LoadResolver.cpp (hunk 1 above)
Comment (TerranCivilian) [reply libyul/optimiser/LoadResolver.cpp:34]: Good catch; fixed.

**Comment by chriseth** [comment]
@TerranCivilian can you squash all the commits into one?

**Comment by TerranCivilian** [comment]
Yes sir, right away sir

**Review by chriseth — APPROVED** [review]

**Pull request merged**